plugins/woocommerce/src/Blocks/BlockTypes/PaymentMethodIcons.php

```diff
@@ -0,0 +1,168 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use WP_Block;
+
+/**
+ * PaymentMethods class.
+ */
+class PaymentMethodIcons extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'payment-method-icons';
+
+	/**
+	 * Get the frontend script handle for this block type.
+	 *
+	 * @param string $key Data to get, or default to everything.
+	 * @return array|string
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Get the frontend style handle for this block type.
+	 *
+	 * @return string[]
+	 */
+	protected function get_block_type_style() {
+		return array_merge( parent::get_block_type_style(), [ 'wc-blocks-packages-style' ] );
+	}
+
+	/**
+	 * Extra data passed through from server to client for block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 */
+	protected function enqueue_data( array $attributes = [] ) {
+		parent::enqueue_data( $attributes );
+		$this->asset_data_registry->add( 'paymentMethodIcons', $this->get_enabled_card_types() );
+		$this->asset_data_registry->add( 'wooPaymentsEnabled', $this->is_woo_payments_enabled() );
```

Comment:
Is this data used anywhere? I see `get_enabled_card_types` and `is_woo_payments_enabled` methods are called elsewhere anyway so we could probably avoid setting the data in registry. Unless there's a purpose for it I'm missing.

Reply:
I added these so we could access this data in the Editor (e.g. through `window.wcSettings.wooPaymentsEnabled`). However, I'm not sure how things like this are usually handled and maybe there's a better way.

Reply:
I ended up removing `wooPaymentsEnabled` and using `getPaymentMethods()` to check if WooPayments is enabled. I'm still using `paymentMethodIcons` to provide the card icons so they're consistently available on both the server and client side.